horreum-backend/src/main/java/io/hyperfoil/tools/horreum/server/SecurityInit.java

```diff
@@ -95,4 +102,29 @@ private void addTeamMembership(UserInfo userInfo, String teamName, TeamRole role
         Optional<Team> storedTeam = Team.find("teamName", teamName).firstResultOptional();
         userInfo.teams.add(new TeamMembership(userInfo, storedTeam.orElseGet(() -> Team.getEntityManager().merge(new Team(teamName))), role));
     }
+
+    // --- //
+
+    /**
+     * Create an admin account if there are no accounts in the system.
+     * The account should be removed once other accounts are created.
+     */
+    @WithRoles(extras = Roles.HORREUM_SYSTEM)
+    void checkInitialAccount(UserBackEnd backend) {
+        long userCount = UserInfo.count();
+        if (userCount == 0) {
+            UserService.NewUser user = new UserService.NewUser();
+            user.user = new UserService.UserData("", INITIAL_ACCOUNT, "Temporary", "Admin", "temp@horreum.com");
+            user.password = new SecureRandom().ints(RANDOM_PASSWORD_LENGTH, '0', 'z' + 1).collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append).toString();
+
+            backend.createUser(user);
+            backend.updateAdministrators(List.of(INITIAL_ACCOUNT));
+            Log.infov("Created temporary admin account: {0} with password {1}", INITIAL_ACCOUNT, user.password);
+
+        } else if (userCount > 1 && !backend.info(List.of(INITIAL_ACCOUNT)).isEmpty()) {
+            Log.warnv("Please remove temporary admin account: {0}", INITIAL_ACCOUNT);
+            // TODO: remove account or add a way to remove accounts
```

Comment:
at the moment there is no way to remove an account. would need to implement that feature.